php/libraries/User.class.inc

```diff
@@ -525,6 +525,55 @@ class User extends UserPermissions
         $this->update($updateArray);
     }
 
+    /**
+     * Use the Pwned Passwords API to determine whether a user is trying
+     * to use a password that has been revealed in public data breaches.
+     *
+     * @param string $password The password to check for pwnage.
+     *
+     * @return bool Whether the password is known to be pwned.
+     */
+    function pwnedPassword(string $password): bool
+    {
+        // TODO Use the HttpClient in LORIS core to cURL more simply.
+
+        /* The pwned passwords API searches a range of password hashes to
+         * protect the actual SHA1 hash from being revealed.
+         * More here: <https://haveibeenpwned.com/API/v2#PwnedPasswords>
+         */
+        $hash       = sha1($password);
+        $hashPrefix = substr(sha1($password), 0, 5);
+        $url        = "https://api.pwnedpasswords.com/range/$hashPrefix";
```

Comment:
Need to add a note in the documentation that need access to outside site. Lots of network admin will prevent outside access from web server to prevent contact to a C&C.

Need to manage timeout in case network admin refuse to open firewall or site is down.
Caveat, some firewall require IP address and do not support FQDN.

Reply:
Sorry for double post to add in the comment, GitHub is quite slow to update a page with latest comment this morning